### PR TITLE
add --omit-publisher to omit publisher part of the path

### PR DESCRIPTION
**Pull request by Wuschli**
I added the option to omit the publisher part of the path to flatten the hierarchy and make the directory more intuitively searchable imho.

The wheel was tested successfully.

Already downloaded files can be moved with `find ./DriveThruRPG/ -maxdepth 2 -mindepth 2 -type d -print0 | xargs -0 -I {} mv -v "{}" ./DriveThruRPG`
Empty publisher folders can be deleted with `find ./DriveThruRPG/ -empty -type d -delete` afterwards.

### CODE REVIEW

**Comment by glujan** [comment]
Hi @Wuschli , thanks for a PR. I like the idea!

I have a minor complaint, though. I think a user should not be able to combine the new settings and `--compatibility-mode` as it might be confusing. Could you make this two options exclusive?

**Comment by glujan** [comment]
You should be able to achieve that with [add_mutually_exclusive_group](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group)

File: drpg/sync.py [hunk 1]
```diff
@@ -126,7 +127,10 @@ def _file_path(self, product: Product, item: DownloadItem) -> Path:
         )
         product_name = _normalize_path_part(product["products_name"], self._compatibility_mode)
         item_name = _normalize_path_part(item["filename"], self._compatibility_mode)
-        return self._library_path / publishers_name / product_name / item_name
+        if  not self._omit_publisher:
```
Comment (glujan) [inline drpg/sync.py:130]: Please fix formatting so the linter does not complain
```suggestion
        if not self._omit_publisher:
```

**Comment by Wuschli** [comment]
thanks for your input. I think I fixed both of your points and also added a new test. Can you please review the test, since I am not proficient with python, especially unit tests.
Cheers, Lukas

**Comment by glujan** [comment]
@Wuschli this looks great, thanks!

I'll take it from here if you don't mind. I'd like to test it locally and will provided any changes if needed.

**Comment by glujan** [comment]
@Wuschli I've released drpg 2024.11.8 with your changes.

**Pull request merged**